src/NuGet.Core/NuGet.Packaging/Signing/Utility/SignatureUtility.cs

```diff
@@ -581,14 +583,7 @@ private static bool IsMatch(X509Certificate2 certificate, EssCertId essCertId)
                 }
             }
 
-            byte[] actualHash;
-
-            using (var hashAlgorithm = CryptoHashUtility.GetSha1HashProvider())
-            {
-                actualHash = hashAlgorithm.ComputeHash(certificate.RawData);
-            }
-
-            return essCertId.CertificateHash.SequenceEqual(actualHash);
+            return essCertId.CertificateHash.Length == HashLength;
```

Comment:
Should we test that an incorrect `essCertId.CertificateHash` is accepted?

Reply:
Thanks. Added two tests: mismatched hash with same length, shorter hash.